src/Features/Core/Portable/GenerateMember/GenerateParameterizedMember/AbstractGenerateMethodService.State.cs

```diff
@@ -159,20 +164,24 @@ internal new class State : AbstractGenerateParameterizedMemberService<TService,
                 else
                 {
                     var typeInference = semanticDocument.Document.GetLanguageService<ITypeInferenceService>();
-                    var delegateType = typeInference.InferDelegateType(semanticModel, SimpleNameOrMemberAccessExpression, cancellationToken);
-                    if (delegateType != null && delegateType.DelegateInvokeMethod != null)
+                    var delegateInvokeMethod = typeInference.InferDelegateType(semanticModel, SimpleNameOrMemberAccessExpression, cancellationToken)?.DelegateInvokeMethod;
+                    if (delegateInvokeMethod != null)
                     {
-                        SignatureInfo = new MethodSignatureInfo(semanticDocument, this, delegateType.DelegateInvokeMethod);
+                        // If we inferred Func/Action here, attempt to create better parameter names than the default
+                        // 'arg1/arg2/arg3' form that the delegate specifies.
+                        var parameterNames = delegateInvokeMethod.ContainingType is { Name: nameof(Action) or nameof(Func<int>), ContainingNamespace.Name: nameof(System) }
```

Comment:
❔ Why restrict this to `Action<T>` and `Func<T, TResult>`? It seems like it would apply to any parameter that has a generic type (`T` in these cases).

➡️ Other delegates with generic parameters might provide more descriptive names for those parameters. The issue in this case arose from the use of `arg` as the parameter name, and not just the fact that the type of `arg` was `T`.

Reply:
Yeah, i can see us broadening this out.  FOr now though, these names are particularly annoying to me :D

Reply:
📝 This is a case where I realized the answer to my question as I was writing it, and rather than remove the question I just included my answer (marked with ➡️) with the question.